components/server/src/workspace/workspace-service.ts

```diff
@@ -61,11 +62,35 @@ export class WorkspaceService {
     async getWorkspace(userId: string, workspaceId: string): Promise<Workspace> {
         await this.auth.checkPermissionOnWorkspace(userId, "access", workspaceId);
 
-        const workspace = await this.db.findById(workspaceId);
-        if (!workspace || !!workspace.softDeleted || workspace.deleted) {
-            throw new ApplicationError(ErrorCodes.NOT_FOUND, "Workspace not found.");
+        return this.doGetWorkspace(workspaceId);
+    }
+
+    async getOwnerToken(userId: string, workspaceId: string): Promise<string> {
+        await this.auth.checkPermissionOnWorkspace(userId, "access", workspaceId);
+
+        // Check: is deleted?
+        await this.doGetWorkspace(workspaceId);
```

Comment:
Looks like you extracted `doGetWorkspace` to avoid the extra call to `checkPermissionOnWorkspace(userId, "access", workspaceId)` ahen call ing the public `getWorkspace` method. It's ok / not a big deal but as we are hitting the check API of spicedb heavily anyway I'd prefer to optimize once we see problems and have data rather than making smaller code compromises here and there to slightly reduce the amount of calls. Just wanted to share no need to take action in this case.

Reply:
This was not about the performance, but for some changes in how checks are done. But that is superfluous now, so removing it. :+1: